IAGrim/Database/DAO/ItemTagDaoImpl.cs

```diff
@@ -37,35 +39,47 @@ class ItemTagDaoImpl : BaseDao<ItemTag>, IItemTagDao {
             Logger.InfoFormat("Stored {0} item tags to database..", items.Count);
         }
 
-
         public IList<ItemTag> GetClassItemTags() {
+            const string namePattern = @"\[ms\](.*)\[fs\](.*)";
+
             using (var session = SessionCreator.OpenStatelessSession()) {
                 using (session.BeginTransaction()) {
                     return session.CreateCriteria<ItemTag>()
                         .Add(Restrictions.Like(nameof(ItemTag.Tag), "tagSkillClassName%"))
                         .List<ItemTag>()
-                        .Select(m => new ItemTag { Name = m.Name, Tag = m.Tag.Replace("tagSkillClassName", "class") })
+                        .Select(m => new ItemTag {
+                            Name = Regex.IsMatch(m.Name, namePattern)
+                                ? Regex.Replace(m.Name, namePattern, "$1/$2")
+                                : m.Name,
+                            Tag = m.Tag.Replace("tagSkillClassName", "class")
+                        })
                         .ToList();
                 }
             }
         }
 
         public ISet<ItemTag> GetValidClassItemTags() {
+            const string gdxClassesPattern = @"tagGDX\dClass(\d+)SkillName00A";
+            const string namePattern = @"\[ms\](.*)\[fs\](.*)";
+
             using (var session = SessionCreator.OpenStatelessSession()) {
                 using (session.BeginTransaction()) {
-                    return session.CreateSQLQuery("SELECT * FROM ItemTag WHERE (Tag LIKE 'tagSkillClassName%' OR Tag LIKE 'tag%Class%SkillName00A') AND LENGTH(Name) > 1")
+                    return session
```

Comment:
IDE not respecting the line length in .editorconfig?

Not the end of the world, but a bit odd.

Reply:
You meant line 69? Can probably cut it if you want.

Reply:
Not an issue, just a curiosity, that it didnt respect the editorconfig.